archimedes_bin/main.cpp

```diff
@@ -1,22 +1,17 @@
 #include <Engine.h>
 #include <Logger.h>
 
+struct MyApp : arch::Application {};
+
 int main() {
 	arch::Logger::init(arch::LogLevel::trace);
 
-	arch::Logger::trace("Hello, {}!", "World");
-	arch::Logger::debug("Hello, {}!", "World");
-	arch::Logger::info("Hello, {}!", "World");
-	arch::Logger::warn("Hello, {}!", "World");
-	arch::Logger::error("Hello, {}!", "World");
-	arch::Logger::critical("Hello, {}!", "World");
-
-	arch::Logger::log(arch::LogLevel::info, "Info log");
-
 	arch::EngineConfig config {
 		600, 480, "Archimedes Test", glm::vec4(0, 0, 0, 0)
 	};
 
-	arch::Engine engine { config };
+	arch::Ref<MyApp> myApp = arch::createRef<MyApp>();
```

Comment:
I think myApp should be either moved to engine, or constructed in-place in engine's constructor, making it as reference is not necessary.

Reply:
We agreed upon this design on last meeting. It should be Ref.